### PR TITLE
statistics: forbid global-stats in stats-ver1 and make analyze options take effect in global-stats

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: forbid global-stats in stats-ver1 and make analyze options take effect in global-stats

### What is changed and how it works?
statistics: forbid global-stats in stats-ver1 and make analyze options take effect in global-stats

### Check List 

Tests 

- Unit test

### Release note 

- statistics: forbid global-stats in stats-ver1 and make analyze options take effect in global-stats

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -366,6 +367,10 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
 		if err != nil {
 			return
 		}
+		statistics.CheckAnalyzeVerOnTable(partitionStats, &statsVer)
+		if statsVer != statistics.Version2 {
+			break // global-stats only support stats-ver2
```
Comment (rebelice) [inline statistics/handle/handle.go:372]: Why don't we return an error here?

**Review by rebelice — COMMENTED** [review]
rest LGTM

**Review by rebelice — APPROVED** [review]

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -405,23 +411,15 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
 		// Because after merging TopN, some numbers will be left.
 		// These remaining topN numbers will be used as a separate bucket for later histogram merging.
 		var popedTopN []statistics.TopNMeta
-		n := uint32(0)
 		for _, topN := range allTopN[i] {
 			if topN == nil {
 				continue
 			}
```
Comment (Reminiscent) [inline statistics/handle/handle.go:417]: This part can be deleted.

Comment (qw4990) [reply statistics/handle/handle.go:417]: You are right. It is deleted.

**Review by Reminiscent — APPROVED** [review]
/LGTM

**Pull request merged**